src/elements/content-sidebar/DocGenSidebar/DocGenSidebar.tsx

```diff
@@ -84,17 +84,25 @@ const DocGenSidebar = ({ getDocGenTags }: Props) => {
         });
 
         return jsonPathsMap;
-    };
+    }, []);
 
-    const loadTags = async () => {
+    const loadTags = useCallback(async (attempts = 10) => {
+        if(attempts <= 0){
+            setIsLoading(false);
+            return
+        }
         setIsLoading(true);
         try {
             const response: DocGenTemplateTagsResponse = await getDocGenTags();
-            if (response && !!response.data) {
+            if(response.message){
+                setTimeout(() => {
+                    loadTags.call(this, attempts - 1);
+                }, 1000);
+            } else if (response && !!response.data) {
                 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                 // @ts-ignore
-                const { data } = response || [];
 
+                const { data } = response || [];
```

Comment:
the array should be an object? but I would remove `|| []` since the `if` check already validates that `response` exists
```suggestion
                const { data } = response || {};
```
or
```suggestion
                const { data } = response;
```